### ISSUE TITLE
z_mean does not change in CVAE CNN

### ISSUE THREAD

**Issue by tuthang102** [PacktPublishing/Advanced-Deep-Learning-with-Keras#17]
After the model CVAE CNN is trained, the values of z_mean encoded on the test set is the same.

**Comment by tuthang102** [comment]
encoder = Model([inputs, y_labels],
                [z_mean, z_log_var, z], 
                name='encoder')

The output of the encoder is z_mean, z_log_var, z. I understand the z_mean is generated by the encoder, the problem I'm having is the values for z_mean are all the same for all the input test data. in CVAE. This is not the case when working with regular VAE.

**Comment by roatienza** [comment]
CVAE (and VAE) encoder `z_mean`  distribution has an objective function to approximate a Gaussian with zero mean and std=1.  On the test set, this can be verified by inserting this code: 

    z, _, _ = encoder.predict([x_test, to_categorical(y_test)],
                              batch_size=batch_size)
    print("shape:", z.shape)
    print("mean:", np.mean(z, axis=0))
    print("std:", np.std(z, axis=0))
    return


Then, it prints (verifying it is not constant but a distribution):

shape: (10000, 2)
mean: [ 0.07939055  0.02497174]
std: [ 0.91762519  0.92036796]

**Comment by tuthang102** [comment]
When I plot the 10000 values, it does not resemble a normal distribution.  All 10000 values are the same.
